docs/eloquent-models/schema-builder.txt

```diff
@@ -0,0 +1,393 @@
+.. _laravel-schema-builder:
+
+==============
+Schema Builder
+==============
+
+.. facet::
+   :name: genre
+   :values: tutorial
+
+.. meta::
+   :keywords: php framework, odm, code example, schema facade, artisan, eloquent, blueprint, artisan, migrate
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+Laravel provides a **facade** to access the schema builder class ``Schema``,
+which lets you create and modify tables. Facades are static interfaces to
+classes that make the syntax more concise and improve testability.
+
+{+odm-short+} supports a subset of the index and collection management methods
+in the Laravel ``Schema`` facade.
+
+To learn more about facades, see `Facades <https://laravel.com/docs/{+laravel-docs-version+}/facades>`__
+in the Laravel documentation.
+
+The following sections describe the Laravel schema builder features available
+in {+odm-short+} and show examples of how to use them:
+
+- :ref:`<laravel-eloquent-migrations>`
+- :ref:`<laravel-eloquent-collection-exists>`
+- :ref:`<laravel-eloquent-indexes>`
+
+.. note::
+
+   {+odm-short+} supports managing indexes and collections, but
+   excludes support for MongoDB JSON schemas for data validation. To learn
+   more about JSON schema validation, see :manual:`Schema Validation </core/schema-validation/>`
+   in the {+server-docs-name+}.
+
+.. _laravel-eloquent-migrations:
+
+Perform Laravel Migrations
+--------------------------
+
+Laravel migrations let you programmatically create, modify, and delete
+your database schema by running methods included in the ``Schema`` facade.
+The following sections explain how to author a migration class when you use
+a MongoDB database and how to run them.
+
+Create a Migration Class
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can create migration classes manually or generate them by using the
+``php artisan make:migration`` command. If you generate them, you must make the
+following changes to perform the schema changes on your MongoDB database:
+
+- Replace the ``Illuminate\Database\Schema\Blueprint`` import with
+  ``MongoDB\Laravel\Schema\Blueprint`` if it is referenced in your migration
+- Use only commands and syntax supported by {+odm-short+}
+
+.. tip::
+
+   If your default database connection is set to anything other than your
+   MongoDB database, update the following items so that Artisan performs
```

Comment:
This reference to "Artisan" is strange to me. The application is a unit (framework, config, classes...), so I'd use the passive form:

> update the following items so that the migration is performed on the correct database

Reply:
Thanks, I'll avoid mentioning Artisan when possible.